dcos_test_utils/diagnostics.py

```diff
@@ -0,0 +1,103 @@
+"""Utilities for running and downloading diagnostics reports
+
+Based on the utilities in dcos/dcos-integration-tests test_dcos_diagnostics.py
+
+This is tested via the test_dcos_diagnostics.py module in the dcos-integration-test module in dcos/dcos
+"""
+
+import datetime
+import logging
+import os
+
+import retrying
+
+from dcos_test_utils.helpers import (
+    ARNodeApiClientMixin,
+    ApiClientSession,
+    RetryCommonHttpErrorsMixin,
+    check_json
+)
+
+log = logging.getLogger(__name__)
+
+
+class Diagnostics(ARNodeApiClientMixin, RetryCommonHttpErrorsMixin, ApiClientSession):
```

Comment:
Would it make sense to make this a subclass of DcosApiSession? Its fairly confusing that `all_slaves` is required just to support the ARNodeApiClientMixin, so it feels like it should be bound to an instantiation that supports that